### PR TITLE
minor code cleanups in layer_builder

### PR DESCRIPTION
**Pull request by turboMaCk**
Reading through layer_builder.rs implementation I found few minor improvements that can be made.

### CODE REVIEW

File: src/layer_builder.rs [hunk 1]
```diff
@@ -79,7 +79,7 @@ where
         settings.set_map_id(map_id);
         settings.set_layer_id(layer_id);
 
-        let mut layer = Layer::new(settings.clone());
+        let mut layer = Layer::new(settings);
```
Comment (turboMaCk) [inline src/layer_builder.rs:82]: use Copy implementation

File: src/layer_builder.rs [hunk 2]
```diff
@@ -155,9 +155,8 @@ where
             ..LayerBundle::default()
         };
 
-        let mut layer = layer_bundle.layer;
+        let layer = layer_bundle.layer;
         let mut transform = layer_bundle.transform;
-        layer.settings.layer_id = layer.settings.layer_id;
```
Comment (turboMaCk) [inline src/layer_builder.rs:160]: this seemed rather redundant

File: src/layer_builder.rs [hunk 3]
```diff
@@ -230,7 +228,7 @@ where
         let morton_tile_index = morton_index(tile_pos);
         if morton_tile_index < self.tiles.capacity() {
             if let Some(tile) = &self.tiles[morton_tile_index].1 {
-                return Ok(&tile);
+                return Ok(tile);
```
Comment (turboMaCk) [inline src/layer_builder.rs:231]: compiler can do this implicitly. I personally don't mind it so I checked with clippy and it complains about this so this updates it to version clippy prefers.

**Review by IceSentry — APPROVED** [review]

**Comment by IceSentry** [comment]
Might as well fix the other clippy issues in other files.

**Review by StarArawn — APPROVED** [review]

**Pull request merged**